bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/tags/SystemTags.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.library.tags;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * System-wide semantic tags
+ *
+ * @author Stefan Triller - Initial Contribution
+ *
+ */
+@NonNullByDefault
+public class SystemTags {
```

Comment:
I had a brief discussion with Frank in the office about that and the idea was to be able to compare the string of the tag from the start towards the end (case insensitive, I agree). So if you have "Purpose:Alarm:Fire" and "Purpose:Alarm:CO", but you are only interested in all alarms you can query for startswith("Purpose:Alarm)... WDYT?

Reply:
Sounds ok, but in the end you would always need to use "string.toLowerCase().startswith" instead of "equalsIgnoreCase" because you can never be sure, if there is some hierarchy.